doc/decisions/header_include.md

```diff
@@ -0,0 +1,123 @@
+# Including Headers
+
+## Problem
+
+In C you can include a header file with
+
+```c
+#include "header.h"
+```
+
+or
+
+```c
+#include <header.h>
+```
+
+In the C standard, section 6.10.2, paragraphs 2 to 4 state:
+
+> 2. A preprocessing directive of the form
+>
+>    ```c
+>    #include <h-char-sequence> new-line
+>    ```
+>
+>    searches a sequence of implementation-defined places for a header identified uniquely by the specified sequence between the `<` and `>` delimiters, and causes the replacement of that directive by the entire contents of the header. How the places are specified or the header identified is implementation-defined.
+>
+> 3. A preprocessing directive of the form
+>
+>    ```c
+>    #include "q-char-sequence" new-line
+>    ```
+>
+>    causes the replacement of that directive by the entire contents of the source file identified by the specified sequence between the `"` delimiters. The named source file is searched for in an implementation-defined manner. If this search is not supported, or if the search fails, the directive is reprocessed as if it read
+>
+>    ```c
+>    #include <h-char-sequence> new-line
+>    ```
+>
+>    with the identical contained sequence (including `>` characters, if any) from the original directive.
+>
+> 4. A preprocessing directive of the form
+>
+>    ```c
+>    #include pp-tokens new-line
+>    ```
+>
+>    (that does not match one of the two previous forms) is permitted. The preprocessing tokens after include in the directive are processed just as in normal text. (Each identifier currently defined as a macro name is replaced by its replacement list of preprocessing tokens.) The directive resulting after all replacements shall match one of the two previous forms. The method by which a sequence of preprocessing tokens between a `<` and a `>` preprocessing token pair or a pair of `"` characters is combined into a single header name preprocessing token is implementation-defined.
+>
+> ### Definitions:
+>
+> - h-char: any member of the source character set except the new-line character and `>`
+> - q-char: any member of the source character set except the new-line character and `"`
+
+In short, the whole search process of the `#include` mechansim is implementation-defined.
+However, because `#include` works (mostly) like a literal copy-paste, deciding between `""` and `<>` within a public header file can affect user code.
```

Comment:
The problem should always describe a problem with Elektra. And it should be important enough, I don't think consistent use of `""` and `<>` is important enough for a decision or a rule in CODING.md.

Reply:
Not really an update here?